src/models/channel.rs

```diff
@@ -142,14 +142,19 @@ impl MainToPeerTask {
                 "make specific peer discovery req"
             }
             MainToPeerTask::TransactionNotification(_) => "transaction notification",
-            MainToPeerTask::Disconnect(_) => "disconnect",
+            MainToPeerTask::Disconnect(..) => "disconnect",
             MainToPeerTask::DisconnectAll() => "disconnect all",
             MainToPeerTask::BlockProposalNotification(_) => "block proposal notification",
         }
         .to_string()
     }
 }
 
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub(crate) enum InternalDisconnectReason {
+    OutOfConnectionCapacity,
+}
+
```

Comment:
What's the motivation for having an enum with one variant? Isn't it cleaner to drop the enum altogether?

Reply:
I'm assuming that this enum will grow in the future. For example, we could add an RPC to terminate some connection, giving rise to a new variant here.

Reply:
That sounds like a useful command to have. However, it might take a while before its priority is high enough to justify implementation and until then we are dragging around needless complexity. Where do you stand on that?

Reply:
It's certainly a trade off. I think building for extensibility is usually a good idea, but I'm not even sure the PR does that. For example, I have now realised that the `DisconnectAll` does not state a reason, which is inconsistent. I have reverted the introduction of `InternalDisconnectReason`, favoring a clean solution now over a maybe-working-maybe-broken extensibility solution.